### ISSUE TITLE
Disable smart filtering

### ISSUE THREAD

**Issue by brunjick** [yajra/laravel-datatables#423]
Is it possible to disable smart filtering on server side? Because, as I could see, datatables doesn't send any parameter for that, if I set
```
$("#example").DataTable({
    ...
    search: {
        smart: false
    }
    ...
})
```

**Comment by yajra** [comment]
@brunjick, good point. We currently don't have way to disable smart search. Currently we have two config for search but setting both to false will still do smart search and will still perform `like '%search%'`.

```php
    'search'          => [
        'case_insensitive' => false,
        'use_wildcards'    => false,
    ],
``` 

Will add this for enhancement to be configurable via config file. We cannot use the js `smart: false` since this data is not being sent to the server as far as I know. Thanks!

```php
    'search'          => [
        'smart' => true
        'case_insensitive' => false,
        'use_wildcards'    => false,
    ],
```

**Comment by yajra** [comment]
@brunjick, please clarify. What do you mean by smart search? Does it mean **exact search** or you wanted to **disable global search**?

My interpretation is exact search but just wanted to make sure before I try to implement this. Thanks!

**Comment by brunjick** [comment]
Yes I mean exact search. More specifically (https://datatables.net/reference/option/search.smart)[official docs here]

**Comment by yajra** [comment]
Thanks for clarifying. Will try to implement this as soon as I can. :+1:

**Comment by yajra** [comment]
@brunjick, I just created a PR for this feature request. Would you please test if the proposal works for you? Thanks!

**Comment by brunjick** [comment]
Yes, I tested and it worked after fixing one problem. When setting where clause on query object
`$this->query->where($column, 'like', $keyword);`
provided column name is already wrapped in backquote (\`) symbol, so we have to remove them, otherwise query builder will escape them and it becomes wrapped in three backquote symbols. So above code becomes something like this:
```$this->query->where(str_replace('`', '', $column), 'like', $keyword);```
I'm not sure if it's perfect way, but it works now.

Also, is there any chance to implement option to define specific columns for disabling/enabling smart search? for example:
```
 Datatables::of($query)
            ->smart(false, ['column1', 'column2'])
            ->make(true);
```

**Comment by yajra** [comment]
Just wondering why you're column has backtick? Laravel automatically adds it when using query builder so I guess it's not needed?

Will check if it's possible to do it on a column level smart search. Thanks!

**Comment by brunjick** [comment]
No, my column doesn't have one
`$column = $this->connection->getQueryGrammar()->wrap($column);`
this line adds backtick in `QueryBuilderEngine#castColumn` method.

I've fixed it for myself, you can take a look at commit on my fork
https://github.com/brunjick/laravel-datatables/commit/2167ff9d1a01f86e3937a6259bfbe56c756032de

**Comment by yajra** [comment]
@brunjick thanks for the hint. I see the issue now and was fixed on this [commit](https://github.com/yajra/laravel-datatables/pull/452/commits/091aae65fa073f2467b7275180c6112f3bf186ba). Can you please try again? 

Column level smart search to follow.

**Comment by brunjick** [comment]
@yajra just one little fix
https://github.com/brunjick/laravel-datatables/commit/aaf5e02bdc52b3dd374b355747bd65877bb3098b

**Comment by yajra** [comment]
@brunjick, thanks for keeping an eagle eye. A bit sleepy last night when I pushed the patch.

Anyways, it's now fixed. Kindly check again and will release this on the next version. Will work on a new PR for a column level smart search. Thanks!

**Comment by brunjick** [comment]
Yes it does work now

**Comment by yajra** [comment]
@brunjick, thanks for the help. Feature released on v6.10.0!

**Comment by amitavmohanty** [comment]
Thank You @yajra for the response, 
I will try to implement it and let you know...

Regards
Amitav
